### PR TITLE
Use data access service to obtain GCP access tokens

### PR DESCRIPTION
**Pull request by bjornandre**

### CODE REVIEW

File: src/main/java/no/ssb/dapla/gcs/token/delegation/BrokerTokenIdentifier.java [hunk 1]
```diff
@@ -109,20 +110,21 @@ public int hashCode() {
             return this;
         }
 
-        public BrokerTokenIdentifier.Builder withNamespace(Text namespace) {
+        public BrokerTokenIdentifier.Builder withNamespacewithNamespace(Text namespace) {
```
Comment (hadrienk) [inline src/main/java/no/ssb/dapla/gcs/token/delegation/BrokerTokenIdentifier.java]: Is this a typo?

Comment (bjornandre) [reply src/main/java/no/ssb/dapla/gcs/token/delegation/BrokerTokenIdentifier.java]: Yes, already fixed

**Comment by kimcs** [comment]
Please use the http transcoded grpc api of data-access, not the self-maintained http api.

File: src/main/java/no/ssb/dapla/service/DataAccessClient.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    private String buildUrl(String format, Object... args) {
+        return this.baseURL + String.format(format, args);
+    }
+
+    public AccessTokenProvider.AccessToken getAccessToken(String userId, String location,
+                                                          AccessTokenRequest.Privilege privilege) {
+        Request request = new Request.Builder()
+                .url(buildUrl("access/token?userId=%s&location=%s&privilege=%s", userId, location, privilege.name()))
```
Comment (kimcs) [inline src/main/java/no/ssb/dapla/service/DataAccessClient.java]: Use transcoded api here! i.e. replace "access/token" with "rpc/DataAccessService/getAccessToken" and put the parameters in the body by first using AccessTokenRequest.newBuilder()....build() and the convert that to json.

Comment (bjornandre) [reply src/main/java/no/ssb/dapla/service/DataAccessClient.java]: Fixed

**Pull request merged**